soh/src/code/z_play.c

```diff
@@ -264,6 +264,18 @@ void GivePlayerRandoRewardRequiem(PlayState* play, RandomizerCheck check) {
     }
 }
 
+void GivePlayerRandoRewardMasterSword(PlayState* play, RandomizerCheck check) {
+    Player* player = GET_PLAYER(play);
+
+    if (gSaveContext.entranceIndex == 0x02CA && LINK_IS_ADULT && !Player_InBlockingCsMode(play, player) &&
+    player != NULL && Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD) && !Flags_GetRandomizerInf(RAND_INF_TOT_MASTER_SWORD)) {
```

Comment:
`player != NULL` should be checked before accessing player in `Player_InBlockingCsMode()`

```suggestion
    if (gSaveContext.entranceIndex == 0x02CA && LINK_IS_ADULT && player != NULL &&
        !Player_InBlockingCsMode(play, player) && Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD) &&
        !Flags_GetRandomizerInf(RAND_INF_TOT_MASTER_SWORD)) {
```